app/src/main/java/org/p2p/wallet/swap/interactor/orca/OrcaSwapInteractor.kt

```diff
@@ -80,15 +82,32 @@ class OrcaSwapInteractor(
                 slippage = slippage.doubleValue,
             )
         } else {
-            swapByFeeRelayer(
-                sourceAddress = fromToken.publicKey,
-                sourceTokenMint = fromToken.mintAddress,
-                destinationAddress = toToken.publicKey,
-                destinationTokenMint = toToken.mintAddress,
-                poolsPair = bestPoolsPair,
-                amount = amount,
-                slippage = slippage.doubleValue,
-            )
+            try {
+                swapByFeeRelayer(
+                    sourceAddress = fromToken.publicKey,
+                    sourceTokenMint = fromToken.mintAddress,
+                    destinationAddress = toToken.publicKey,
+                    destinationTokenMint = toToken.mintAddress,
+                    poolsPair = bestPoolsPair,
+                    amount = amount,
+                    slippage = slippage.doubleValue,
+                )
```

Comment:
Could be nested function

Reply:
how?

Reply:
you can try and extract the whole call to swapByFeeRelayer to a variable

Reply:
how?
in words it is obvious to me that it would be cool to reuse it
but i do not know how in best way to do this!

Reply:
```kotlin 
        suspend fun swapByFeeRelayer() = swapByFeeRelayer(
            sourceAddress = fromToken.publicKey,
            sourceTokenMint = fromToken.mintAddress,
            destinationAddress = toToken.publicKey,
            destinationTokenMint = toToken.mintAddress,
            poolsPair = bestPoolsPair,
            amount = amount,
            slippage = slippage.doubleValue,
        )

        return if (shouldUseNativeSwap(feePayerToken.mintAddress)) {
            swapNative(
                poolsPair = bestPoolsPair,
                sourceAddress = fromToken.publicKey,
                destinationAddress = toToken.publicKey,
                amount = amount,
                slippage = slippage.doubleValue,
            )
        } else {
            try {
                swapByFeeRelayer()
            } catch (serverError: ServerException) {
                if (serverError.errorCode == ErrorCode.INVALID_BLOCKHASH) {
                    // if something not ok with BLOCKHASH we can retry transaction with a new one
                    swapByFeeRelayer()
```